src/SlmQueue/Queue/AbstractQueue.php

```diff
@@ -65,6 +65,14 @@ public function unserializeJob($string, array $metadata = array())
         $metadata += $data['metadata'];
         $content  =  unserialize($data['content']);
 
+        if (isset($data['metadata']['__jobchain__'])) {
+            $self = $this; // needed for php5.3 (?)
```

Comment:
I'm not against changing requirements. But shouldn't we do that in a separate PR?